internal/services/gateway/action/auth.go

```diff
@@ -211,8 +222,21 @@ func (h *ActionHandler) CanAuthUserDoRunActions(ctx context.Context, groupType s
 	if err != nil {
 		return false, "", errors.Wrapf(err, "failed to determine ownership")
 	}
-	if !isProjectOwner {
+	if actionType == actionTypeRunAction && ownerType == cstypes.ObjectKindOrg {
+		userID := common.CurrentUserID(ctx)
+		isUserOrgMember, err := h.IsUserOrgMember(ctx, userID, ownerID)
+		if err != nil {
+			return false, "", errors.Wrapf(err, "failed to determine ownership")
+		}
+		if isProjectOwner || (isUserOrgMember && p.MembersCanPerformRunActions) {
+			return true, refID, nil
+		}
+
 		return false, "", nil
+	} else {
+		if !isProjectOwner {
+			return false, "", nil
+		}
 	}
 	return true, refID, nil
```

Comment:
You have both a default return and a return at the end of the first branch
You should remove the default return or remove the return at the of the first branch.